File: algorithms/bit_manipulation/check_sign.java
class check_sign {
  public static boolean is_positive(int num) {
    return num >> (Integer.BYTES * 8 - 1) == 0;
  }
  public static void main(String[] args) {
    System.out.println(is_positive(10));
    System.out.println(is_positive(-10));
  }
}

File: algorithms/bit_manipulation/check_sign.py
import sys

def is_positive(num):
  return num >> (sys.getsizeof(int) * 8 - 1) == 0

print("%d\n" % (is_positive(10)))
print("%d\n" % (is_positive(2323)))
print("%d\n" % (is_positive(1001)))

print("%d\n" % (is_positive(-10)))
print("%d\n" % (is_positive(-11)))
print("%d\n" % (is_positive(-13)))